server/src/main/java/org/elasticsearch/index/snapshots/blobstore/SlicedInputStream.java

```diff
@@ -99,4 +116,49 @@ public final int available() throws IOException {
         return stream == null ? 0 : stream.available();
     }
 
+    @Override
+    public boolean markSupported() {
+        return true;
+    }
+
+    @Override
+    public void mark(int readLimit) {
+        // We ignore readLimit since openSlice() can re-open previous InputStreams, and we can skip as many bytes as we'd like.
+        // According to JDK documentation, marking a closed InputStream should have no effect.
+        if (markSupported() && isClosed() == false && numSlices > 0) {
+            if (initialized) {
+                markedSlice = nextSlice - 1;
+                markedSliceOffset = currentSliceOffset;
+            } else {
+                markedSlice = 0;
+                markedSliceOffset = 0;
+            }
+        }
+    }
+
+    @Override
+    public void reset() throws IOException {
+        if (markSupported()) {
+            if (isClosed()) {
+                throw new IOException("reset called on a closed stream");
+            } else if (numSlices > 0) {
+                if (markedSlice < 0 || markedSliceOffset < 0) {
+                    throw new IOException("Mark has not been set");
+                }
+
+                nextSlice = markedSlice;
+                nextStream();
+
+                // We do not call the SlicedInputStream's skipNBytes but call skipNBytes directly on the returned stream, to ensure that
+                // the skip is performed on the marked slice and no other slices are involved. This may help uncover any bugs.
+                final InputStream stream = currentStream();
```

Comment:
Nit:
```suggestion
                // We do not call the SlicedInputStream's skipNBytes but call skipNBytes directly on the returned stream, to ensure that
                // the skip is performed on the marked slice and no other slices are involved. This may help uncover any bugs.
                final InputStream stream = nextStream();
```
?